tests/test_types.py

```diff
@@ -6685,3 +6685,49 @@ class Foo(BaseModel):
     # validation error raised bc foo field uses strict mode
     with pytest.raises(ValidationError):
         Foo(foo='1')
+
+
+def test_complex_field():
```

Comment:
Tests look good so far, though I think you're right, we'll want to support the string `1+2j` format :)